From: Arjan van de Ven <arjan@linux.intel.com>
Subject: Fix Atom C6 mwait value and target residency

The intel_idle driver has the wrong mwait value for the C6 power state,
and in addition, had a too short target residency.
This patch corrects both values.

Signed-off-by: Arjan van de Ven <arjan@linux.intel.com>

---
 drivers/idle/intel_idle.c |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

Index: linux-2.6.35/drivers/idle/intel_idle.c
===================================================================
--- linux-2.6.35.orig/drivers/idle/intel_idle.c
+++ linux-2.6.35/drivers/idle/intel_idle.c
@@ -159,13 +159,13 @@ static struct cpuidle_state atom_cstates
 	{ /* MWAIT C5 */ },
 	{ /* MWAIT C6 */
 		.name = "ATM-C6",
-		.desc = "MWAIT 0x40",
-		.driver_data = (void *) 0x40,
+		.desc = "MWAIT 0x52",
+		.driver_data = (void *) 0x52,
 		.flags = CPUIDLE_FLAG_TIME_VALID,
 		.exit_latency = 200,
 		.power_usage = 150,
-		.target_residency = 800,
-		.enter = NULL },	/* disabled */
+		.target_residency = 4000,
+		.enter = &intel_idle },
 };
 
 /*
@@ -309,6 +309,7 @@ static int intel_idle_probe(void)
 
 	case 0x1C:	/* 28 - Atom Processor */
 	case 0x26:	/* 38 - Lincroft Atom Processor */
+	case 0x27:	/* 39 - Atom variant */
 		lapic_timer_reliable_states = (1 << 2) | (1 << 1); /* C2, C1 */
 		cpuidle_state_table = atom_cstates;
 		choose_substate = choose_zero_substate;
